var/spack/repos/builtin/packages/ecp-data-vis-sdk/package.py

```diff
@@ -15,17 +15,22 @@ def dav_sdk_depends_on(spec, when=None, propagate=None):
     # ie. A +c ~b -> A
     spec = Spec(spec).name
 
-    if "+" in when and len(when.split()) == 1:
-        when_not = when.replace("+", "~")
-        # If the package is in the spec tree then it must
-        # be enabled in the SDK.
-        conflicts(when_not, "^" + spec)
+    # If the package is in the spec tree then it must be enabled in the SDK.
+    if "+" in when:
+        _when_variants = when.strip("+").split("+")
+        if any(tok in when for tok in ["~", "="]):
+            tty.error("Bad token in when clause, only positive boolean tokens allowed")
+
+        for variant in _when_variants:
+            conflicts("~" + variant, when="^" + spec)
 
     # Skip if there is nothing to propagate
     if not propagate:
         return
 
-    # Map the propagated variants to the dependency variant
+    # Map the propagated variants to the dependency variant.  Some packages may need
+    # overrides to propagate a dependency as something else, e.g., {"visit": "libsim"}.
+    # Most call-sites will just use a list.
```

Comment:
FYI before we forget about this comment I just want to sneak it in on one of these PRs